receiver/sapmreceiver/trace_receiver.go

```diff
@@ -75,36 +74,37 @@ func (sr *sapmReceiver) handleRequest(ctx context.Context, req *http.Request) er
 		return err
 	}
 
+	ctx = obsreport.ReceiverContext(ctx, sr.config.Name(), "http", "")
+	ctx = obsreport.StartTraceDataReceiveOp(ctx, sr.config.Name(), "http")
+	numSpans := 0
+
 	// process sapm batches
 	for _, batch := range sapm.Batches {
+
 		// convert the jager batches to OCProto
-		td, err := jaegertranslator.ProtoBatchToOCProto(*batch)
+		// TODO: the translator never returns error change the function signature.
+		td, _ := jaegertranslator.ProtoBatchToOCProto(*batch)
```

Comment:
May be we should keep err handling until the translator changes it's signature?

Reply:
Using this fact we can simplify a bit the code if we keep it we should handle it different to get correct metrics.